bundles/org.openhab.binding.smartthings/src/main/java/org/openhab/binding/smartthings/internal/handler/SmartthingsThingHandler.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smartthings.internal.handler;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.status.ConfigStatusMessage;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.ConfigStatusThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.smartthings.internal.SmartthingsBindingConstants;
+import org.openhab.binding.smartthings.internal.SmartthingsHandlerFactory;
+import org.openhab.binding.smartthings.internal.converter.SmartthingsConverter;
+import org.openhab.binding.smartthings.internal.dto.SmartthingsStateData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Bob Raker - Initial contribution
+ */
+@NonNullByDefault
+public class SmartthingsThingHandler extends ConfigStatusThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SmartthingsThingHandler.class);
+
+    private SmartthingsThingConfig config;
+    private String smartthingsName;
+    private SmartthingsHandlerFactory smartthingsHandlerFactory;
+    private Map<ChannelUID, SmartthingsConverter> converters = new HashMap<ChannelUID, SmartthingsConverter>();
+
+    private final String smartthingsConverterName = "smartthings-converter";
+
+    public SmartthingsThingHandler(Thing thing, SmartthingsHandlerFactory smartthingsHandlerFactory) {
+        super(thing);
+        this.smartthingsHandlerFactory = smartthingsHandlerFactory;
+        smartthingsName = ""; // Initialize here so it can be NonNull but it should always get a value in initialize()
+        config = getThing().getConfiguration().as(SmartthingsThingConfig.class);
```

Comment:
The configuration might not be valid at construction, so instead use a default config instance to make the nonnull annotation happy.
```suggestion
        config = new SmartthingsThingConfig();
```

Reply:
Done